README.md

```diff
@@ -27,6 +27,118 @@ WARNING: If the configuration failed or if the firewall configuration limits
 access to the machine in a bad way, it is most likely be needed to get
 physical access to the machine to fix the issue.
 
+Ansible Facts
+-------------
+
+## Gathering firewall ansible facts
+
+To gather the firewall system role's ansible facts, 
+call the system role with no arguments e.g.
+```yaml
+vars:
+  firewall:
+```
+
+Another option is to gather a more detailed version of the
+ansible facts by using the detailed argument e.g.
+```yaml
+vars:
+  firewall:
+    detailed: true
+```
+
+## Available ansible facts
+
+### firewall_config
+
+This ansible fact shows the permanent configuration of
+of firewalld on the managed node in dictionary format, 
+broken into two subdictionaries: default and custom. 
+Each dictionaries have the keys:
+- `zones`
+- `services`
+- `icmptypes`
+- `helpers`
+- `ipsets`
+- `policies` (if supported by remote host's firewalld installation)
+
+Each of the keys contains a list of elements present in 
+permanent configuration for each respective option.
+
+custom will have a list of subdictionaries for each key,
+providing a more detailed description.
+default will have only the names of each setting,
+unless the detailed option is supplied, in which case
+it will be structured in the same manner as custom.
+
+JSON representation of the structure of firewall_config fact:
+```json
+{
+  "default": {...},
+  "custom": {...},
+}
+```
+
+#### default
+
+The default subdictionary of firewall_config contains the default
+configuration for the managed node's firewalld configuration. 
+This subdictionary only changes with changes to the managed node's 
+firewalld installation.
+
+default without detailed parameter set to true
+```json
+"default": {
+  "zones": [...],
+  "services": [...],
+  "icmptypes": [...],
+  "helpers": [...],
+  "ipsets": [...],
+  "policies": [...],
+}
+```
+
+default when parameter set to true
+```json
+"default": {
+  "zones": [{}...],
+  "services": [{}...],
+  "icmptypes": [{}...],
+  "helpers": [{}...],
+  "ipsets": [{}...],
+  "policies": [{}...],
+}
+```
+
```

Comment:
Is this enough to accurately represent the structure of the ansible fact? I definitely cannot provide a full example for a detailed firewall_config.default, because I am certain that approaches 10,000 lines if shown in full.

Reply:
Right.  Just enough to give the reader some rough idea about what it will look like.

Really?  10,000 lines?  Then I'm wondering how a user will actually use this at all, much less print it - something like `debug: msg: firewall config {{ firewall_config | to_nice_json }}` is likely to blow up the ansible log

Reply:
It does, thats why i supress logs and why i make `detailed: false` the default

Reply:
just checked ansible log, 10000 was a little high, ~4600 is a lot more accurate